commands/pipelines/setup.js

```diff
@@ -0,0 +1,231 @@
+  const params = {
+    source_blob: { url: archiveURL },
+    app: { name }
+  }
+
+  if (organization) {
+    params.app.organization = organization
+  } else {
+    params.app.personal = true
+  }
+
+  return api.createAppSetup(heroku, params).then((setup) => {
+    return api.postCoupling(heroku, pipeline.id, setup.app.id, stage).then(() => {
+      return setup.app
+    })
+  })
+}
+
+function* getNameAndRepo (args) {
+  const answers = yield prompt([{
+    type: 'input',
+    name: 'name',
+    message: 'Pipeline name',
+    when () { return !args.name }
+  }, {
+    type: 'input',
+    name: 'repo',
+    message: 'GitHub repository to connect to (e.g. rails/rails)',
+    when () { return !args.repo },
+    validate (input) {
+      if (input.match(REPO_REGEX)) return true
+      return 'Must be in the format organization/rep  o'
+    }
+  }])
+
+  return Object.assign(answers, args)
+}
+
+function* getSettings (branch) {
+  return yield prompt([{
+    type: 'confirm',
+    name: 'auto_deploy',
+    message: `Automatically deploy the ${branch} branch to staging?`
+  }, {
+    type: 'confirm',
+    name: 'wait_for_ci',
+    message: `Wait for CI to pass before deploying the ${branch} branch to staging?`,
+    when (answers) { return answers.auto_deploy }
+  }, {
+    type: 'confirm',
+    name: 'pull_requests.enabled',
+    message: 'Enable review apps?'
+  }, {
+    type: 'confirm',
+    name: 'pull_requests.auto_deploy',
+    message: 'Automatically create review apps for every PR?',
+    when (answers) { return answers.pull_requests.enabled }
+  }, {
+    type: 'confirm',
+    name: 'pull_requests.auto_destroy',
+    message: 'Automatically destroy idle review apps after 5 days?',
+    when (answers) { return answers.pull_requests.enabled }
+  }])
+}
+
+function* hasCIFlag (heroku) {
+  let hasFlag
+  try {
+    hasFlag = (yield api.getAccountFeature(heroku, 'ci')).enabled
+  } catch (error) {
+    hasFlag = false
+  }
+  return hasFlag
+}
+
+function* getCISettings (organization) {
+  const settings = yield prompt([{
+    type: 'confirm',
+    name: 'ci',
+    message: 'Enable automatic Heroku CI test runs?'
+  }])
+
+  if (settings.ci && organization) {
+    settings.organization = organization
+  }
+
+  return settings
+}
+
+function setupPipeline (token, app, settings, pipelineID, ciSettings = {}) {
+  const promises = [kolkrabbi.updateAppLink(token, app, settings)]
+
+  if (ciSettings.ci) {
+    promises.push(
+      kolkrabbi.updatePipelineRepository(token, pipelineID, ciSettings)
+    )
+  }
+
+  return Promise.all(promises).then(([appLink]) => {
+    return appLink
+  }, (error) => {
+    cli.error(error.response.body.message)
+  })
+}
+
+module.exports = {
+  topic: 'pipelines',
+  command: 'setup',
+  description: 'bootstrap a new pipeline with common settings',
+  help: `Example:
+
+  heroku pipelines:setup example githuborg/reponame -o example-org
```

Comment:
I chose not to because we don't currently need it. I think it would be easy to extend this to support that syntax if/when we actually want it.